sync_labels.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python3
+import sys
+import os
+import argparse
+from github import Github, GithubException
+
+"""
+Sync Github Labels
+
+Synchronize labels idempotently between two Github repositories.
+"""
+
+
+def sync_labels(src_repo, dest_repo, dry_run, delete):
+    """
+    Main subroutine: sync labels between source/destination Github repos
+
+    :param src_repo: full repo name for source of labels
+    :param dest_repo: full repo name for destination of label actions
+    :param dry_run: do a dry run, do not perform the actions
+    :param delete: delete labels that are in destination but not in source
+    """
+
+    # Get PyGithub objects representing repos
+    g = Github(get_access_token())
+    try:
+        s = g.get_repo(src_repo)
+    except GithubException:
+        raise Exception(f"Source repo is not a repo: {src_repo}")
+    try:
+        d = g.get_repo(dest_repo)
+    except GithubException:
+        raise Exception(f"Destination repo is not a repo: {dest_repo}")
+
+    # Create list of labels to add/delete
+    src_labels = list(s.get_labels())
+    src_label_names = set([j.name for j in src_labels])
+    dest_labels = list(d.get_labels())
+    dest_label_names = set([j.name for j in dest_labels])
+    dest_add = src_label_names - dest_label_names
+    dest_del = dest_label_names - src_label_names
+
+    # Show summary of what actions will be taken, get confirmation
+    print(f"The following labels will be created in {dest_repo}:")
+    for lab in sorted(list(dest_add)):
+        print(f" [+] {lab}")
+
+    if delete:
+        print(f"The following labels will be deleted in {dest_repo}:")
+        for lab in sorted(list(dest_del)):
+            print(f" [-] {lab}")
+
+    # Add new labels
+    for name in dest_add:
+        print(f"Creating label {name} in destination repo {dest_repo}...")
```

Comment:
I think the follow-up print statements add a lot of unneeded noise. Removing the ellipses here along with the follow-up messages would be equally clear and more concise. This is a suggestion though.

Reply:
Addressed in e796840 (recording success/failure to report at end, instead of as we go)